presto-tpcds/src/main/java/com/facebook/presto/tpcds/TpcdsConnectorFactory.java

```diff
@@ -109,6 +110,16 @@ private int getSplitsPerNode(Map<String, String> properties)
         }
     }
 
+    private boolean toggleCharToVarchar(Map<String, String> properties)
```

Comment:
It seems like this option is only valid in Native code as the Java RecordSet always uses false for it.

If we expect that the only use-case for this property is for native code temporarily and should be removed after Prestissimo starts supporting char type, then we can remove this config and just generate the TableMetadata correctly depending on isNativeExecutionEnabled().

Reply:
The TPC-DS standard specifies the column as char and Presto Java supports it correctly as well. We are kind of regressing by allowing the option to use varchar in the schema instead.

The Native engine is working around its limitation of types and using varchar for these fields. I feel this is very explicit knowledge for tools/pipelines using Native engine. They sign up for this behavior when they decide to use the tpc-ds connector with native engine.

@pdabre12 : Do we have a notion of native only/java only session config with side-car enabled ? This is a good candidate for a native only config I feel.

@majetideepak : Do you feel that the varchar/char(n) work will come along in the near term ? If yes, then we can use the correct type in the TPC-DS connector from the beginning itself.

Reply:
> We are kind of regressing by allowing the option to use varchar in the schema instead.

We can simply throw an error for Java that varchar is not supported. Don't have to add support. Similarly, we will throw an error for Native that char is not supported until we support it.

> They sign up for this behavior when they decide to use the tpc-ds connector with native engine. 

They will only know this by reading the documentation or realize this after generating some TPC-DS data. The config will give them an advance notice.

> Do you feel that the varchar/char(n) work will come along in the near term 

This work is getting closer to completion. It does make sense to wait for this to complete instead of adding workarounds.

Reply:
@aditi-pandit We should be able to do that, it is still an ongoing effort : https://github.com/prestodb/presto/pull/23968/

Reply:
> We can simply throw an error for Java that varchar is not supported. Don't have to add support. Similarly, we will throw an error for Native that char is not supported until we support it.

This would mean throwing for almost all the tables because most of them have a column with char data type..